zapcore/field.go

```diff
@@ -160,6 +160,23 @@ func (f Field) AddTo(enc ObjectEncoder) {
 	case NamespaceType:
 		enc.OpenNamespace(f.Key)
 	case StringerType:
+		defer func() {
```

Comment:
Given the function is pretty large and this is a large block, it'd be better to move the logic into a separate `encodeStringer` function

Reply:
Done.  We call `encodeStringer` for `StringerType` which contains a call to `catchPanic`.